distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Filter_Condition_Helpers.enso

```diff
@@ -67,5 +81,24 @@ make_filter_column source_column filter_condition on_problems = case filter_cond
         if is_nan_column.is_error then is_infinite_column.not else
             (is_infinite_column || is_nan_column).not
     # Vector
-    Is_In values -> source_column.is_in values
-    Not_In values -> source_column.is_in values . not
+    Is_In values ->
+        warn_on_nothing_in_comparison_vector filter_condition values <|
+            source_column.is_in values
+    Not_In values ->
+        warn_on_nothing_in_comparison_vector filter_condition values <|
+            source_column.is_in values . not
+
+## Attach a warning if the provided value is `Nothing`.
+warn_on_nothing_in_comparison : Filter_Condition -> Any -> Any -> Any
+warn_on_nothing_in_comparison filter_condition value ~action =
+    case value of
+        Nothing -> Warning.attach (Nothing_Value_In_Filter_Condition.Error filter_condition) action
+        _ -> action
+
+## Attach a warning if the provided value is a `Vector` that contains `Nothing`.
+warn_on_nothing_in_comparison_vector : Filter_Condition -> Vector Any -> Any -> Any
+warn_on_nothing_in_comparison_vector filter_condition values ~action =
+    case values of
+        _ : Vector ->
+            values.fold action (flip (warn_on_nothing_in_comparison filter_condition))
+        _ -> action
```

Comment:
If a column is passed in, shouldn't we also raise the warning if it contains nulls?

Reply:
If the column is costly to compute, are we concerned about the overhead?

Reply:
I assume the intention is that we wanted to detect explicit `Nothing` values used to construct `Filter_Condition`s, not the presence of Nothing in the data itself.

Reply:
> I assume the intention is that we wanted to detect explicit `Nothing` values used to construct `Filter_Condition`s, not the presence of Nothing in the data itself.

Then I'm not sure if we should warn about the Vector case either.

I think we should be consistent, as otherwise we are building user expectations (_I can expect a warning_) and then breaking them (_If I provide a Column instead of the Vector, there's no warning anymore_)

Reply:
I think the Vector case is different. That is a direct argument to Filter_Condition.Is_In, and we are warning the user that they are explicitly using a value that cannot match anything. A Column is generally from the user's source-of-truth dataset, and could be very large and often contain Nothing.

My understanding is that we are providing a warning about what was explicitly passed in to a Filter_Condition constructor, not about what the user's dataset might contain.

In addition, scanning the Column for Nothing requires that it be computed, which could be costly.

Reply:
Fair, from that point of view the current behaviour seems OK.